File: 프로그래머스/1/12926. 시저 암호/시저 암호.js
function solution(s, n) {
    var answer = '';
    
    for(let i=0; i<s.length; i++){
        if(s.charCodeAt(i)==32){
            answer += s[i];
        }else{
        if(s.charCodeAt(i) >= 97){
            if(s.charCodeAt(i)+n > 122){
                answer += String.fromCharCode(s.charCodeAt(i)+n-26);
            }else{
                answer += String.fromCharCode(s.charCodeAt(i)+n);
            }
                
        }else{
            if(s.charCodeAt(i)+n > 90){
                answer += String.fromCharCode(s.charCodeAt(i)+n-26);
            }else{
                answer += String.fromCharCode(s.charCodeAt(i)+n);
            }
        }
        }
    }
    return answer;
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
class Solution {
    public String solution(String s, int n) {
        String answer = "";

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);

            // 공백은 그대로
            if (c == ' ') {
                answer += c;
            }
            // 대문자
            else if (c >= 'A' && c <= 'Z') {
                if (c + n > 'Z') {
                    answer += (char)(c + n - 26);
                } else {
                    answer += (char)(c + n);
                }
            }
            // 소문자
            else if (c >= 'a' && c <= 'z') {
                if (c + n > 'z') {
                    answer += (char)(c + n - 26);
                } else {
                    answer += (char)(c + n);
                }
            }
        }

        return answer;
    }
}